helloworld/BUILD

```diff
@@ -1,33 +1,37 @@
 # Copyright 2020 Pants project contributors.
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-# `name` defaults to the name of this directory, i.e., `helloworld`.
-pex_binary(
-    sources=["main.py"],
-    entry_point="helloworld.main",
-)
-
+# This target sets the metadata for all the Python non-test files in this directory.
+#
+# * `name` defaults to the name of this directory, i.e., `helloworld`.
+# * `sources` defaults to ['*.py', '!*_test.py', '!test_*.py', '!conftest.py'].
+# * Pants cannot infer dependencies on resources targets, so we explicitly add it.
 python_library(
-  name="awslambda_lib",
-  sources=["awslambda.py"],
+    dependencies=[":config_file"],
 )
 
-# Note: This has sdist dependencies, so it must be built on Linux.
-python_awslambda(
-    name="helloworld-awslambda",
-    dependencies=[":awslambda_lib"],
-    handler="helloworld.awslambda:handler",
-    runtime="python3.7",
+# This target sets the metadata for the JSON file in this directory, which allows for other targets
```

Comment:
It doesn't really set any metadata though. I mean it cannot - resource targets have no metadata fields AFAICR. It literally exists just so we can depend on it. Which again means we should rethink targets, but for now the comment should probably be streamlined.

Reply:
> It literally exists just so we can depend on it.

Yeah, true.

> Which again means we should rethink targets

Strong +1

Reply:
Well, it exists as a way to depend on a glob of files, which you would otherwise need to list manually. Resources collections are very rarely individual files.